pandas/tests/frame/test_dtypes.py

```diff
@@ -552,6 +553,10 @@ def test_astype_to_incorrect_datetimelike(self, unit):
         with pytest.raises(TypeError, match=msg):
             df.astype(dtype)
 
+    def test_astype_sparsedtype_empty_dataframe(self):
```

Comment:
since the same issue occurs with all extension array dtypes can you add a test to pandas\tests\extension\base\dtype.py instead.

Reply:
yes!